BlueprintUI/Sources/Layout/LayoutWriter.swift

```diff
@@ -193,6 +200,51 @@ extension LayoutWriter {
 }
 
 
+extension LayoutWriter.Context {
+
+    /// The current phase of the layout event: `.measurement` or `.layout`.
+    public enum LayoutPhase: Equatable {
+
+        /// The element is being measured.
+        case measurement
+
+        /// The element is being laid out with a known size.
+        case layout
+
+        /// Returns the provided value based on if a measurement or layout is occurring.
+        public func ifMeasuring<Value>(
+            _ ifMeasuring: () -> Value,
+            ifLayout: () -> Value
+        ) -> Value {
+            switch self {
+            case .measurement: return ifMeasuring()
+            case .layout: return ifLayout()
+            }
+        }
+
+        /// Returns the provided value based on if a measurement or layout is occurring.
+        public func ifMeasuring<Value>(
+            _ ifMeasuring: @autoclosure () -> Value,
+            ifLayout: @autoclosure () -> Value
+        ) -> Value {
+            switch self {
+            case .measurement: return ifMeasuring()
+            case .layout: return ifLayout()
+            }
+        }
```

Comment:
I'm fine with adding these if we're finding them useful, but IMO it's easier and more clear to just use a switch where you need it.

E.g. declare an in-scope var where you need it
```swift
var size: CGSize {
    switch context.phase {
    case .measuring:
        return contentSize
    case .layout:
        return context.size.maximum
    }
}
```

use a closure that's executed immediately
```swift
size: {
    switch context.phase {
    case .measuring:
        return contentSize
    case .layout:
        return context.size.maximum
    }
}()
```

or provide a `var isMeasuring: Bool` convenience:
```swift
size: context.phase.isMeasuring ? contentSize : context.size.maximum
```

Reply:
I updated this to be a bit more useful – the layout callback now takes in the size, so consumers don't need to "implicitly" convert the constraint to a size.